code/golang/where.go

```diff
@@ -24,14 +24,14 @@ type PartitionedArgs struct {
 
 func PartitionedArgsFromWheres(wheres []*ir.Where) (out PartitionedArgs) {
 	for _, where := range wheres {
-		if where.Right != nil {
+		if !where.Right.HasPlaceholder() {
 			continue
 		}
 
 		arg := ArgFromWhere(where)
 		out.AllArgs = append(out.AllArgs, arg)
 
-		if where.Nullable() {
+		if where.NeedsCondition() {
```

Comment:
i think this changed the semantics of partitioned args... the arg may or may not be nullable and that's different from if the where is rendered with a condition, right? 

maybe it's ok because each where maps to exactly 0 or 1 arg, and for the where to need a condition, the arg must be nullable. is that the case?